tests/openssl_moretest/host/CSchema_checker.cpp

```diff
@@ -0,0 +1,359 @@
+// Copyright (c) Open Enclave SDK contributors.
+// Licensed under the MIT License.
+
+#include <assert.h>
+#include <stdio.h>
+#include <stdlib.h>
+
+#include "CSchema_checker.h"
+#include "include_openssl.h"
+
+CSchemaChecker::CSchemaChecker(t_openssl_schema* schema, uint schema_size)
+{
+    m_schema_idx = 0;
+    m_schema_num_apis = schema_size;
+
+    memset(m_p1.p, 0, sizeof(m_p1.p));
+    memset(m_p2.p, 0, sizeof(m_p2.p));
+    if (schema)
+    {
+        m_schema = (t_openssl_schema*)new t_openssl_schema[schema_size];
```

Comment:
Will C++ throw or return NULL on failure?  http://www.cplusplus.com/reference/new/nothrow/ implies you have to use (std::nothrow) to tell new to not throw so that you can throw your own exception in line 22, but I'm not sure.  (Same question applies to line 179 and anywhere else you use 'new')

Reply:
I think it is OK to throw an exception to handle the failure in constructor.

Reply:
Do you have any suggestion on that? Thanks.

Reply:
@mingweishih  and @dthaler , Please help me find a better solution to handle the failure in constructor. Or could you run Bors to close this PR?

Reply:
@jinghe-INTC I think the question is that does `new` returns `nullptr` on failure (i.e., whether the check of `if (!m_schema)` in line 21 actually work). If that works, I think it should address @dthaler's concerns.

Reply:
So if you agree, I will take `new (std::nothrow)`_`type`_ for each of the usage of `new`, and then check if the return value is `nullptr`.

Reply:
@dthaler Can you check again if this addresses your concern?